### PR TITLE
[monorepo] feat: add Jenkins jobs

### PR DESCRIPTION
**Pull request by Wayonb**
problem: currently Jenkins is not enable
solution: 1. enable Jenkins jobs and build scripts
              2. fix linting failures
              3. enable dependabot

### CODE REVIEW

**Comment by Wayonb** [comment]
Website build is currently broken so did not add scripts.
nem-docker maybe add should add linting for the shell scripts?

File: .github/workflows/auto-merge.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  contents: write
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1.1.1
```
Comment (Jaguar0625) [inline .github/workflows/auto-merge.yaml]: 1.4 ?

File: .github/buildConfiguration.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    path: explorer/frontend
+
+  - name: NEM Nano Wallet
+    path: nanowallet
+
+  - name: NEM Open API
+    path: nem-openapi
+
+  - name: NEM Nemesis Generator
+    path: nemesis-generator
```
Comment (Jaguar0625) [inline .github/buildConfiguration.yaml:12]: anything for website or docs?

File: explorer/frontend/scripts/ci/test.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+set -ex
+
+CI=true npm run test:jenkins
+
```
Comment (Jaguar0625) [inline explorer/frontend/scripts/ci/test.sh]: extra newline

File: nanowallet/scripts/ci/test.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -ex
+
+if [ "$1" = "code-coverage" ];
+then
+    npm run test:jenkins
+else
+    npm run test
+fi
```
Comment (Jaguar0625) [inline nanowallet/scripts/ci/test.sh]: tab indent?

File: nanowallet/Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,6 @@
+defaultCiPipeline {
+	platform = ['ubuntu']
+	ciBuildDockerfile = 'node10.Dockerfile'
+
+	packageId = 'nem-nanaowallet'
+}
```
Comment (Jaguar0625) [inline nanowallet/Jenkinsfile:6]: no linter ?

File: nanowallet/Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,6 @@
+defaultCiPipeline {
+	platform = ['ubuntu']
+	ciBuildDockerfile = 'node10.Dockerfile'
+
+	packageId = 'nem-nanaowallet'
```
Comment (Jaguar0625) [inline nanowallet/Jenkinsfile]: nano

File: nanowallet/Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,6 @@
+defaultCiPipeline {
+	platform = ['ubuntu']
+	ciBuildDockerfile = 'node10.Dockerfile'
```
Comment (Jaguar0625) [inline nanowallet/Jenkinsfile:3]: why node10 instead of javascript?

File: nem-openapi/spec/schemas/AllNodesDTO.yml [hunk 8]
```diff
@@ -10,19 +10,19 @@ properties:
         type: array
```
Comment (Jaguar0625) [inline nem-openapi/spec/schemas/AllNodesDTO.yml:10]: i think we've been using yaml extension everywhere, so might want to rename these (separate change)

File: nemesis-generator/scripts/ci/test.sh [hunk 9]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -ex
+
+TEST_RUNNER=$([ "$1" = "code-coverage" ] && echo "coverage run --append" || echo "python3")
+${TEST_RUNNER} -m configuration_generator --count 20 --seed 100000000000 --network-name testnet --output ./nemesis.yaml --accounts-output ./user.yaml
+${TEST_RUNNER} -m generator -i ./nemesis.yaml -o nemesis.bin
+${TEST_RUNNER} -m node_configuration_generator --accounts-file ./user.yaml --nodes-file ./resources/nodes.yaml --nemesis-file ./nemesis.yaml --seed ./nemesis.bin --network-friendly-name ship --output-path ./output
```
Comment (Jaguar0625) [inline nemesis-generator/scripts/ci/test.sh]: consider updating to match examples in README or vice-versa

File: .github/buildConfiguration.yaml (hunk 2 above)
Comment (Wayonb) [reply .github/buildConfiguration.yaml:12]: website is broken due to some dependency conflict and did not want to create a broken job.
docs, I am not sure what to add yet, still looking into this.
I will look into these in another PR.

File: nanowallet/Jenkinsfile (hunk 7 above)
Comment (Wayonb) [reply nanowallet/Jenkinsfile:3]: NanoWallet requires node 10 to build.  Javascript image is currently at node 16.

Comment (Jaguar0625) [reply nanowallet/Jenkinsfile:3]: do you know what its incompatibility is with a more current version of node?

Comment (Wayonb) [reply nanowallet/Jenkinsfile:3]: Its using ``gulp-cli`` for tasks which seems to only with the older version of nodes.
There are a lot of depreciated modules due to node 10 requirements.

File: nanowallet/Jenkinsfile (hunk 5 above)
Comment (Wayonb) [reply nanowallet/Jenkinsfile:6]: I tried to add linting and hitting a log of errors(~3K).  most seems to be valid so will need through them.
This is an example.
```
/miscell/nanowallet/src/app/modules/voting/polls/polls.controller.js
  169:9   error  'Promise' is not defined       no-undef
  171:18  error  'd' is defined but never used  no-unused-vars
  193:19  error  'e' is defined but never used  no-unused-vars
  223:19  error  'e' is defined but never used  no-unused-vars
  327:26  error  'e' is defined but never used  no-unused-vars
  351:19  error  'e' is defined but never used  no-unused-vars
  370:23  error  'e' is defined but never used  no-unused-vars
```

File: nanowallet/Jenkinsfile (hunk 7 above)
Comment (Wayonb) [reply nanowallet/Jenkinsfile:3]: I also asked about updating the node version in discord - https://discord.com/channels/856325968096133191/865180563433979904/1100437664227921940

**Review by Jaguar0625 — APPROVED** [review]

**Pull request merged**